### ISSUE TITLE
Headlamp goes blank when viewing Ingresses due to port number being undefined

### ISSUE THREAD

**Issue by linhtv-salefronts** [headlamp-k8s/headlamp#1323]
When I open Ingresses menu, the Headlamp sitesgoes blank and show errors regarding the port number.
It was showing **TypeError: Cannot read properties of undefined (reading 'toString')** for the *port* constant.

```
function RulesDisplay(props: { ingress: Ingress }) {
  const { ingress } = props;

  const rulesText = React.useMemo(() => {
    const rules = ingress.getRules();
    let labels: string[] = [];

    rules.forEach(({ http }) => {
      const text = http.paths.map(({ path, backend }) => {
        const service = backend.service.name;
        const port = backend.service.port.number; <~ This is the problem
        return `${path} 🞂 ${!!service ? service + ':' + port.toString() : port.toString()}`;
      });
      labels = labels.concat(text);
    });

    return labels;
  }, [ingress]);

  return <LabelListItem labels={rulesText} />;
}
```

I assume that it's because almost all of my backend is using **port.name** instead of **port.number**
```
backend:
    service:
      name: headlamp
      port:
        name: http
```

It also shows an error in API stream but I'm not sure if it's the consequence of the preceeding error
```
"/apis/networking.k8s.io/v1/ingresses?watch=1&resourceVersion=5220024"
```

**Comment by SquupS** [comment]
Hi @illume,

I just had the same issue like described using version 0.19.0, for both, in-luster and client application

On top there is the same "going blank" behaviour for StorageVolumes. But I am not sure as of now, what is causing that.

**Comment by illume** [comment]
@SquupS Ok, thanks for the report.

**Comment by linhtv-salefronts** [comment]
>  I'm interested to know if Headlamp 0.19.0 works for you or not.

0.19.0? In the [Headlamp helm chart](https://artifacthub.io/packages/helm/headlamp/headlamp), the latest chart version is 0.14.0 and that is what I'm using. It shows the Application version is 0.19.0, but when installing helm chart you have to use the chart version and not the application version. Could it be that you guys forget to update the chart version to match the application version?

**Comment by illume** [comment]
Yes, sorry... it seems the helm chart version and headlamp version are not matching. But it seems you already tried with the latest Headlamp 0.19.0, and someone else was able to reproduce.

It looks like the issue was accepted to be fixed in a Headlamp 0.19.1 release (not the chart version).
